frontend/src/rauta.tsx

```diff
@@ -123,6 +130,18 @@ export const makeRouter = <C extends Config, >(config: C): RouterLib => {
         }
     };
 
+    const shouldPreventNav = (listeners: { listener: BeforeNavigationListener }[]): boolean => {
+        for (const { listener } of listeners) {
+            let prevent = false;
+            listener(() => prevent = true);
```

Comment:
That's a good question. I thought about it a couple times. And it would certainly be easier in our code. But is that what one wants? Especially if the listener are just called "before nav listener"? Calling a passed function is more explicit right? Mhh. What do you think?

Reply:
One could also make them return `"prevent" | null` or something like that. There is also the whole "JS returns `undefined` when you don't explicitly `return`, and `undefined` is false-y" thing, but that should be fixable with a proper return type.

But yes, maybe a naked `boolean` is too implicit. :thinking: I won't insist on this change, but what do you think about a more explicit return type, maybe?

Reply:
I changed it now. Lemme know whatya think